module/VuFind/src/VuFind/Role/PermissionProvider/User.php

```diff
@@ -96,8 +97,11 @@ public function getPermissions($options)
                     $pattern = '/' . $pattern . '/';
                 }
 
-                if (preg_match($pattern, $user[$attribute])) {
-                    return ['loggedin'];
+                if (method_exists($user, 'get' . ucfirst($attribute))) {
+                    $userValue = call_user_func([$user, 'get' . ucfirst($attribute)]);
```

Comment:
We might need special cases for the `cat_id` and `cat_username` attributes. Maybe something like:

```
$methodMap = ['cat_id' => 'getCatId', 'cat_username' => 'getCatUsername'];
$method = $methodMap[$attribute] ?? 'get' . ucfirst($attribute);
if (method_exists($user, $method)) {
    $userValue = $user->$method();
    ...
```

Reply:
Additionally, on a side note: at the top of this function, the user is retrieved using `$this->auth->getIdentity()`. I believe this should be changed to `$this->auth->getUserObject()` since the method is making use of the full user object and not just the basic user identity. This code should have been updated a long time ago, and I suspect it got missed because we were using the wrong retrieval method. Might as well fix it while we're here! (As of today, both methods return the same object, since the old user object implements both the identity and entity interfaces... but if in future we split up those roles, using the right method will become a lot more important!).